### DISCUSSION TITLE
Feature Request - Distance based activation/deactivation to save performance

### DISCUSSION THREAD

**Discussion by lordrex34** [pafuhana1213/KawaiiPhysics#125] [Ideas - 要望・アイデア]
I recommend an option to activate/deactivate the Kawaii Physics nodes based on Distance (in meters probably).

For example if I set "KawaiiPhysicsDistanceLimit = 100" then the system would disable if the distance is higher than 100M. When the distance decreases to 99M it activates again.

**Comment by pafuhana1213** [comment]
Hi @lordrex34 ,
Do you know LOD option of anim node?  This should help in most cases, and has the added benefit of working with other LOD elements.

![image](https://github.com/pafuhana1213/KawaiiPhysics/assets/8957600/7f62edfb-6112-468c-a3ce-06b084404b3a)

thanks!

**Reply by lordrex34** [reply]
Thanks I understand that - though what I really meant is:
* I use kawaii physics for Hair & Cloth
* I want to stop Hair physics after distance 2000
* I want to stop Cloth physics after distance 1000

Can this variation be done via the LOD Threshold  @pafuhana1213 ?

**Reply by pafuhana1213** [reply]
It is possible depending on the LOD settings of the SkeletalMesh. However, it is a bit tedious to set it up, so I recommend updating the Alpha of the KawaiiPhysics node according to the distance. This method is flexible and versatile, so it should solve almost all problems.

**Reply by jtbiddle** [reply]
Does the processing stop (and save resources) if the alpha is zero?

**Reply by pafuhana1213** [reply]
Hi @jtbiddle 

The initialization process runs, but other update and evaluation processes do not run. 
It's in Japanese, but I've researched it and will share the results.
https://pafuhana1213.hatenablog.com/entry/2024/08/19/153447

thanks!
